ui/src/app/views/app/component.ts

```diff
@@ -102,4 +105,30 @@ export class AppComponent implements OnInit {
     }
   }
 
+  /**
+   * Setting the Google Analytics ID here.
+   * This first injects Google's gtag script via iife, then secondarily defines
+   * the global gtag function.
+   */
+  private setGTagManager() {
+    const s = this.doc.createElement('script');
+    s.type = 'text/javascript';
+    s.innerHTML =
+      '(function(w,d,s,l,i){' +
+        'w[l]=w[l]||[];' +
+        'var f=d.getElementsByTagName(s)[0];' +
+        'var j=d.createElement(s);' +
+        'var dl=l!=\'dataLayer\'?\'&l=\'+l:\'\';' +
+        'j.async=true;' +
+        'j.src=\'https://www.googletagmanager.com/gtag/js?id=\'+i+dl;' +
+        'f.parentNode.insertBefore(j,f);' +
+      '})' +
+      '(window, document, \'script\', \'dataLayer\', \'' + environment.gaId + '\');' +
```

Comment:
Why not just inline the function above, as we only call it here, once? (Not sure it matters, just curious.)

Reply:
I did try that, but it doesn't as the function is then scoped in the iife (will expand on that in other comment). It would have made the code a bit shorter if it worked.